channeldb/invoices.go

```diff
@@ -1259,6 +1268,13 @@ func serializeHtlcs(w io.Writer, htlcs map[CircuitKey]*InvoiceHTLC) error {
 			)
 			records = append(records, setIDRecord)
 		}
+		if htlc.Preimage != nil {
```

Comment:
It is not present for accepted and cancelled htlcs? (add comment?)

Reply:
(currently) it is for keysend payments

Reply:
updated so that preimages on AMP payments must be set when moving to Accepted or Settled